File: app/src/main/java/cn/com/woong/readhub/ui/topic/topicdetail/TopicDetailContract.java
package cn.com.woong.readhub.ui.topic.topicdetail;

import cn.com.woong.readhub.base.BaseContract;
import cn.com.woong.readhub.bean.TopicMo;

/**
 * Created by wong on 2018/4/22.
 */

public interface TopicDetailContract {
    interface View extends BaseContract.BaseView {
        void updateTopicDetail(TopicMo topicMo);
    }

    interface Presenter extends BaseContract.BasePresenter<View> {
        void getTopicDetail(String topicId);
    }
}

File: app/src/main/java/cn/com/woong/readhub/ui/topic/topicdetail/TopicDetailContract.kt
package cn.com.woong.readhub.ui.topic.topicdetail

import cn.com.woong.readhub.base.BaseContract
import cn.com.woong.readhub.bean.TopicDetailMo

interface TopicDetailContract {
    interface View : BaseContract.IView {
        fun updateTopicDetail(topicDetailMo: TopicDetailMo)
    }

    interface Presenter : BaseContract.IPresenter<View> {
        fun getTopicDetail(topicId: String)
    }
}